util/travis_compiled_push.sh

```diff
@@ -69,10 +69,9 @@ if [[ "$TRAVIS_COMMIT_MESSAGE" != *"[skip build]"* ]] ; then
 	# rm -f compiled/*.hex
 
 	# ignore errors here
-	for file in ../qmk_firmware/keyboards/*/keymaps/*/*_default.hex; do mv -v "$file" "compiled/${file##*/}" || true; done
-	for file in ../qmk_firmware/keyboards/*/*/keymaps/*/*_default.hex; do mv -v "$file" "compiled/${file##*/}" || true; done
-	for file in ../qmk_firmware/keyboards/*/*/*/keymaps/*/*_default.hex; do mv -v "$file" "compiled/${file##*/}" || true; done
-	for file in ../qmk_firmware/keyboards/*/*/*/*/keymaps/*/*_default.hex; do mv -v "$file" "compiled/${file##*/}" || true; done
+  # In theory, this is more flexible, and will allow for additional expansion of additional types of files and other names
+  mv -t compiled ../qmk_firmware/keyboards/*(*|*/*|*/*/*|*/*/*/*|*/*/*/*/*)/keymaps/*/*_default.*(hex|bin) || true
```

Comment:
```suggestion
  mv -t compiled ../qmk_firmware/keyboards/*(/*|/*/*|/*/*/*|/*/*/*/*)/keymaps/*/*_default.*(hex|bin) || true
```

Reply:
I think that will exclude the base folder.   But I'm not sure/will have to check.

Reply:
Unfortunately, neither appears to work for anything but the first sub-directory.

So unless somebody more skilled with this stuff (eg, not me) has a suggestion, this may be better:
```suggestion
  mv -t compiled ../qmk_firmware/keyboards/*/keymaps/*/*_default.*(hex|bin) || true
  mv -t compiled ../qmk_firmware/keyboards/*/*/keymaps/*/*_default.*(hex|bin) || true
  mv -t compiled ../qmk_firmware/keyboards/*/*/*/keymaps/*/*_default.*(hex|bin) || true
  mv -t compiled ../qmk_firmware/keyboards/*/*/*/*/keymaps/*/*_default.*(hex|bin) || true
  mv -t compiled ../qmk_firmware/keyboards/*/*/*/*/*/keymaps/*/*_default.*(hex|bin) || true
```

Reply:
Also, the reason for this change was so that we can get both the bin and hex files in one go, without having to have 4-5 commands for each format (so 8 or 10 commands here).

Also, this makes it more modular, so we could add other keymaps to the compiled list, such as VIA keymaps.

Reply:
Maybe something like this then?

```suggestion
  find ../qmk_firmware/keyboards -name \*_default.(hex|bin) | grep /keymaps/ | xargs mv -t compiled || true
```

Reply:
It doesn't like `(hex|bin)`. 

```
bash: syntax error near unexpected token `('
```

Adding ?, *,  ^  doesn't help.

Reply:
@vomindoraan Since you're good with this stuff, from what I've seen, any suggestions?

Reply:
We can just pass -name twice, see my edited suggestion.

Reply:
That gives me "missing file operand" :(